3-projects/cyf-ecommerce-api/server.js

```diff
@@ -0,0 +1,267 @@
+const express = require("express");
+const app = express();
+app.use(express.json());
+
+const { Pool } = require("pg");
+
+const pool = new Pool({
+  user: "asif",
+  host: "localhost",
+  database: "cyf_ecommerce",
+  password: "786god",
+  port: 5432,
+});
+
+app.get("/customers", function (req, res) {
+  pool
+    .query("SELECT * FROM customers")
+    .then((result) => {
+      if (result) {
+        res.status(200).json(result.rows);
+      } else res.status(400).send("Couldn't retrieve");
+    })
+    .catch((e) => console.error(e));
+});
+app.get("/customers/:customerId/orders", function (req, res) {
+  const customerId = req.params.customerId;
+
+  const query =
+    "SELECT o.order_reference, o.order_date, p.product_name, pa.unit_price, s.supplier_name, oi.quantity  FROM orders o INNER JOIN order_items oi ON o.id = oi.order_id INNER JOIN products p ON oi.product_id = p.id INNER JOIN product_availability pa ON oi.product_id = pa.prod_id INNER JOIN suppliers s ON oi.supplier_id = s.id WHERE customer_id = $1";
+
+  pool
+    .query(query, [customerId])
+    .then((result) => res.json(result.rows))
+    .catch((e) => console.error(e));
+});
+app.post("/customers/:customerId/orders", function (req, res) {
+  const newCustomerID = req.params.customerId;
+  const newOrderDate = req.body.order_date;
+  const NewOrderRef = req.body.order_reference;
+
+  pool
+    .query("SELECT * FROM customers WHERE customers.id=$1", [newCustomerID])
+    .then((result) => {
+      if (result.rows.length === 0) {
+        return res.status(400).send("No customer with this id exists!");
+      } else {
+        const query =
+          "INSERT INTO orders (order_date, order_reference, customer_id)VALUES ($1, $2, $3)";
+        pool
+          .query(query, [newOrderDate, NewOrderRef, newCustomerID])
+          .then(() => res.send("new order created!"))
+          .catch((e) => console.error(e));
+      }
+    });
+});
+
+app.get("/customers/:customerId", function (req, res) {
+  let customerId = req.params.customerId;
+  pool
+    .query("SELECT * FROM customers where customers.id =$1", [
+      Number(customerId),
+    ])
+    .then((result) => {
+      if (result) res.status(200).json(result.rows);
+      else res.status(400).send("Couldn't retrieve");
```

Comment:
Maybe also check for an empty list and return a  404 and a message if there are no results.